mesonbuild/interpreter/interpreter.py

```diff
@@ -3279,6 +3279,18 @@ def build_target_decorator_caller(self, node, args, kwargs):
                     outputs.update(o)
 
         kwargs['include_directories'] = self.extract_incdirs(kwargs)
+
+        if isinstance(targetclass, build.Executable):
+            if kwargs['gui_app'] is not None:
+                if kwargs['win_subsystem'] is not None:
+                    raise InvalidArguments.from_node(
+                        'Executable got both "gui_app", and "win_subsystem" arguments, which are mutually exclusive',
+                        node=node)
+                if kwargs['gui_app']:
```

Comment:
This `if` looks to be always true to me

Reply:
gui_app is `Optional[bool]`, so it shoudl be either `true` or `false` there. It intentionally falls through in the `false` case so that the the `if kwargs['win_subsystem'] is None` case is the only place we're setting win_subsystem to console